chcemvediet/apps/anonymization/management/commands/attachment_anonymization.py

```diff
@@ -0,0 +1,78 @@
+from optparse import make_option
+
+import magic
+from django.core.files.base import ContentFile
+from django.core.management.base import BaseCommand, CommandError
+
+from poleno.attachments.models import Attachment
+from poleno.utils.misc import squeeze
+from chcemvediet.apps.anonymization.models import AttachmentFinalization
+from chcemvediet.apps.inforequests.models import Action
+
+
+class Command(BaseCommand):
+    args = u'attachment [file]'
+    help = squeeze(u"""
+            Creates anonymization for the specified Attachment. By default, the file path is read
+            from stdin. File path can be explicitly passed as an command argument. Anonymization
+            created this way will be marked as successful. Only one successful anonymization can be
+            assigned to the Attachment.
+            """)
+
+    option_list = BaseCommand.option_list + (
+        make_option(u'--content_type',
+                    help=squeeze(u"""
+                            Content type of file, e.g. "application/pdf". Automatically computed if
+                            not specified.
+                            """)
+                    ),
+        make_option(u'--debug',
+                    default=u'',
+                    help=u'Debug message to the newly created anonymization. Empty by default.'
+                    ),
+        make_option(u'--force',
+                    action=u'store_true',
+                    help=squeeze(u"""
+                            The command refuses to anonymize attachment if a successful
+                            anonymization already exists. This flag disables this check. Deletes all
+                            existing successful anonymizations and creates new one. Unsuccessful
+                            anonymizations will stay unaffected.
+                            """)
+                    ),
+    )
+
+    def handle(self, *args, **options):
+        if not args:
+            raise CommandError(u'attachment_anonymization takes at least 1 argument (0 given).')
+        if len(args) > 2:
+            raise CommandError(
+                u'attachment_anonymization takes at most 2 arguments ({} given).'.format(len(args))
+                )
+        pk = args[0]
+        filename = args[1] if len(args) == 2 else raw_input(u'File:')
+        if not filename:
+            raise CommandError(u'Missing source file name.')
+        with open(filename, u'rb') as file:
```

Comment:
@viliambalaz Zislo by sa ohandlovat, co ak `open(filename, u'rb')` failne. Napriklad ak subor neexistuje, user nema pravo ho citat a podobne. Teraz to padne na error typu `IOError: [Errno 2] No such file or directory: 'foo'` a podobne. Zislo by sa, aby to skoncilo na `CommandError` s rozumnou hlaskou.

Reply:
@Nechal som hlasku "File {} does not exist." a to aj v pripade ak subor existuje, len pouzivatel nema prava na citanie.
Este moze nastat error "ValueError: invalid literal for int()". Treba osetrit tiez?

Reply:
@viliambalaz 
> nechal som hlasku "File {} does not exist." a to aj v pripade ak subor existuje, len pouzivatel nema prava na citanie.

Neda sa z exception raisnutym z `open` vycitat skutocny dovod. A userovi poslat `CommandError` s hlaskou `Could not open file: ...`, kde `...` bude skutocny dovod z `open`.

>  Este moze nastat error "ValueError: invalid literal for int()" ak je zadany `attachment_id` v zlom formate . Treba osetrit tiez?

Pre `attachment_id` staci vo vsetkych pripadoch jeden error, ze taky attachment neexistuje. Ale ano treba to osetrit, ze ak user zada nejake blbosti, tak nech to vzdy padne na `CommandError` s rozumnou hlaskou. Nie na stacktrace.

Reply:
:ok:

Reply:
@martinmacko47 priklad "[Errno 2] No such file or directory". To 1[Errno {}]` je v poriadku? Nechcem zasahovat a menit error hlasku ktora je z `IoError`. Jednoducho som len vypisal to co tam je. Avsak moze byt zavadzajuce, ze cislo erroru sa vstahuje na command `attachment_anonymization` nie na IOerror.

Reply:
@viliambalaz Ano, kludne to `[Errno {}]` moze ostat. Dolezite je, aby to bolo zabalene do `CommandError`, nech to neskonci stacktrasom.